### PR TITLE
Feature/protists

### PR DESCRIPTION
**Pull request by CristiGuijarro**
## Description

_Compara team get the privilege of running protists compara for e107._

**Related JIRA tickets:**
- ENSCOMPARASW-5026
- ENSCOMPARASW-5027
- ENSCOMPARASW-5029

#### Change 1
- New `conf/protists` directory
- `allowed_species.json` populated with protist species, kept under threshold limit and recycled from previous release
- New `mlss_conf.xml` for now just for the `default` collection and protein trees - to be added to in e108/e109 for alignments
- Placeholder for `production_reg_conf.pl` with predicted servers, and expected dbs.

#### Change 2
- Fix to `all-housekeeping/division_config.t` so that test is skipped if no genomes are listed in `mlss_conf.xml` but are in `allowed_species.json` - this was not necessary before because where alignments are involved, there is always a species specified, and the pan division is excepted because it does not have an `allowed_species.json`.

## Testing
_**_Travis._**_

## Notes
_This total work is not complete, but it is functional as is for purpose and ready for branching and large enough for now._

### CODE REVIEW

**Review by dthybert — APPROVED** [review]

**Review by twalsh-ebi — COMMENTED** [review]
It's always good to see another division moving in, and assuming the PipeConfig modules will follow, it all looks grand to me.

I just have some questions:

- The existing Compara `allowed_species.json` files are (mostly) sorted alphabetically. Could that be done in this case? It would make it easier to search for a particular species, and would reduce the chances of a mistake in an update.
- In the `jira_recurrent_tickets.json` file, is it necessary that `mysql-ens-compara-prod-6` be used for `PrepareMasterDatabaseForRelease_conf`? If not, could this be changed to something generic, such as `mysql-ens-compara-prod-X`?

**Comment by CristiGuijarro** [comment]
> It's always good to see another division moving in, and assuming the PipeConfig modules will follow, it all looks grand to me.
> 
> I just have some questions:
> 
> * The existing Compara `allowed_species.json` files are (mostly) sorted alphabetically. Could that be done in this case? It would make it easier to search for a particular species, and would reduce the chances of a mistake in an update.
> * In the `jira_recurrent_tickets.json` file, is it necessary that `mysql-ens-compara-prod-6` be used for `PrepareMasterDatabaseForRelease_conf`? If not, could this be changed to something generic, such as `mysql-ens-compara-prod-X`?

Yes, they should be sorted alphabetically, I forgot that :+1: 
And actually, I also missed the server speciciation, it should be `mysql-ens-compara-prod-8` I think, thanks!

File: conf/protists/allowed_species.json [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    "thalassiosira_oceanica_gca_000296195",
+    "thalassiosira_pseudonana",
+    "thecamonas_trahens_atcc_50062_gca_000142905",
+    "theileria_equi_strain_wa_gca_000342415",
+    "theileria_orientalis_strain_shintoku_gca_000740895",
+    "theileria_parva_gca_000165365",
+    "thraustotheca_clavata_gca_002081575",
+    "tieghemostelium_lacteum_gca_001606155",
+    "toxoplasma_gondii_veg_gca_000150015",
+    "toxoplasma_gondii",
```
Comment (twalsh-ebi) [inline conf/protists/allowed_species.json:130]: A suggestion as optional as it is pedantic:
```suggestion
    "toxoplasma_gondii",
    "toxoplasma_gondii_veg_gca_000150015",
```

**Review by twalsh-ebi — APPROVED** [review]
I have one minor suggestion, but am approving regardless of whether you choose to apply it.

**Pull request merged**